frontend/src/stores/ui.ts

```diff
@@ -133,16 +133,30 @@ export const useUiStore = defineStore("ui", {
      * Given a list of key value pairs of UI state parameters and their states,
      * populate the store state to match the cookie.
      *
+     * Since the cookies are passed through `JSON.parse()`, they can have the
+     * wrong types. This function resets cookies to the default value if their
+     * type is incorrect.
+     *
      * @param cookies - mapping of UI state parameters and their states.
      */
     initFromCookies(cookies: OpenverseCookieState) {
-      this.updateBreakpoint(cookies.uiBreakpoint ?? this.breakpoint)
-      this.isFilterDismissed = cookies.uiIsFilterDismissed ?? false
-      this.isMobileUa = cookies.uiIsMobileUa ?? false
+      let breakpoint = this.breakpoint
+      if (
+        cookies.uiBreakpoint &&
+        Object.keys(ALL_SCREEN_SIZES).includes(cookies.uiBreakpoint)
+      )
+        breakpoint = cookies.uiBreakpoint
+      this.updateBreakpoint(breakpoint)
+      if (typeof cookies.uiIsFilterDismissed === "boolean")
+        this.isFilterDismissed = cookies.uiIsFilterDismissed
+      this.isMobileUa = false
```

Comment:
While the other keys go back to their default values from the store, `isMobileUa` does not (it's set to `false` here but the default is `true`). 

I don't know what the reason for this discrepancy is but this discrepant behaviour is unit-tested so I kept it as is in this PR.